crates/types/src/vote.rs

```diff
@@ -255,7 +319,448 @@ pub trait Accumulator<T, U>: Sized {
     fn append(self, val: T) -> Either<Self, U>;
 }
 
+/// Accumulator trait used to accumulate votes into an `AssembledSignature`
+pub trait Accumulator2<
```

Comment:
Do we want to rename this `Accumulator2` and the old `Accumulator` to something like `Accumulator_basic_vote` and `Accumulator_other_votes`? So that their functionalities are more readable. But this is not a hurry thing I think.

Reply:
I agree!  I put that task in in this issue: https://github.com/EspressoSystems/HotShot/issues/1760. I will work on this issue next! I just wanted to get this merged in a timely fashion since it will cause conflicts with other peoples work.